File: src/leetcode/Solution26.java
package leetcode;

public class Solution26 {
    public int removeDuplicates(int[] nums) {
        if (nums == null || nums.length < 1) {
            return 0;
        }

        int fv = nums[0];
        int index = 1;
        for (int i = 1; i < nums.length; i++) {
            if (nums[i] > fv) {
                nums[index++] = fv = nums[i];
            }
        }

        return index;
    }
}

File: src/leetcode/kotlin/Solution26.kt
package leetcode.kotlin

class Solution26 {
    fun removeDuplicates(nums: IntArray): Int {
        var idx = 0
        for (i in 1 until nums.size) {
            if (nums[idx] != nums[i]) nums[++idx] = nums[i]
        }
        return idx + 1
    }
}